### PR TITLE
Refactor HTTP request handling into HttpAdapter class

### PR DESCRIPTION
**Pull request by ankapone**
- Implement the `HttpAdapter` class to encapsulate the logic for converting Telegram requests to HTTP requests.
- Update `PsrTelegramClient` to use `HttpAdapter` for handling requests.
- Enhance unit tests to verify the functionality of `HttpAdapter`.

### CODE REVIEW

**Comment by vjik** [comment]
What is the purpose of these changes? Just refactoring?

**Comment by ankapone** [comment]
The separation of responsibilities between the client (who actually makes the request) and the builder of that request, and, as a result, the simplification of testing. Well, you could say it's just a refactoring.

File: src/Client/HttpAdapter.php [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+use Http\Message\MultipartStream\MultipartStreamBuilder;
+use Psr\Http\Message\RequestFactoryInterface;
+use Psr\Http\Message\RequestInterface;
+use Psr\Http\Message\StreamFactoryInterface;
+use Vjik\TelegramBot\Api\Request\HttpMethod;
+use Vjik\TelegramBot\Api\Request\TelegramRequestInterface;
+use Vjik\TelegramBot\Api\Type\InputFile;
+
+final readonly class HttpAdapter
```
Comment (vjik) [inline src/Client/HttpAdapter.php:13]:
```suggestion
/**
 * @internal
 */
final readonly class PsrRequestFactory
```

File: src/Client/PsrTelegramClient.php
Comment (vjik) [inline src/Client/PsrTelegramClient.php:1]: Add tests for this class also.

**Review by vjik — CHANGES_REQUESTED** [review]
Good refactoring, but it needs a few fixes.

**Pull request closed without merging**